src/Trace.php

```diff
@@ -18,6 +19,28 @@ public function __construct(
 
     public static function fromEvent(PhpWarningTriggered|PhpNoticeTriggered|PhpDeprecationTriggered $event): self
     {
-        return new self($event->message(), $event->file(), $event->line(), $event->wasSuppressed());
+        $issueStatus = match (true) {
+            $event instanceof PhpWarningTriggered => TestStatus::Warning,
+            $event instanceof PhpNoticeTriggered => TestStatus::Notice,
+            $event instanceof PhpDeprecationTriggered => TestStatus::Deprecated,
+        };
+        return new self($issueStatus, $event->message(), $event->file(), $event->line(), $event->wasSuppressed());
+    }
+
+    /**
+     * Key to identify identical issues, using the same rules as PHPUnit.
+     *
+     * @see \PHPUnit\TestRunner\TestResult\Collector::issueId
+     */
+    public function getIssueId(): string
+    {
+        return sha1(
```

Comment:
What is the need to hash this? This just seems like wasted CPU cycles and unnecessary obfuscation (making it harder to debug program state) to me.

Reply:
I traded RAM for CPU. I was fairly undecided, I'll just get rid of it.